src/basic/memfd-util.c

```diff
@@ -20,6 +20,17 @@
 #include "string-util.h"
 #include "utf8.h"
 
+int memfd_create_wrapper(const char *name, unsigned mode) {
+        int mfd;
+
+        mfd = memfd_create(name, mode);
+
+        if (mfd == -1 && errno == EINVAL)
+                mfd = memfd_create(name, mode & ~(MFD_EXEC | MFD_NOEXEC_SEAL));
+
+        return mfd;
```

Comment:
Please return negative errno on failure:
```c
int memfd_create_wrapper(const char *name, unsigned mode) {
        int mfd;

        mfd = RET_NERRNO(memfd_create(name, mode));
        if (mfd != -EINVAL)
                return mfd;

        return RET_NERRNO(memfd_create(name, mode & ~(MFD_EXEC | MFD_NOEXEC_SEAL)));
}
```

Reply:
The goal was to keep this API-compatible with the libc memfd_create()

Reply:
please, for our own code, always return the negative error code on failure instead of -1 like glibc does. in a way this is compatible, as any code that does the usual `if (r < 0)` check will work either way.

We generally try to program defensively, i.e. not add extra pitfalls that a function implemented by us doesn't follow our usual coding style

Reply:
can we do the fallback logic only if either of these two bits are actually set? seems pointless otherwise

Reply:
Done, thanks for the clarifications and pointers!